### DISCUSSION TITLE
Dayna SCSILINK Wi-Fi problems ...

### DISCUSSION THREAD

**Discussion by franklukas** [BlueSCSI/BlueSCSI-v2#165] [Help] [answered]
I need help ...

With my Macintosh IIsi and System 7.1 the HDD part works fine but Wi-Fi do not work. Logging in via the ini does not work. I have tried everything I could find on the web and nothing works. With the Accessory from ...
https://jcs.org/bluescsi
... it works well.

```
----------- bluescsi.ini ----
[SCSI]
System=Generic
EnableSCSI2=1 ; On
EnableParity=1 ; On
Debug=1 ; On
WiFiSSID=Fritz!Box-Frank
WiFiPassword=18266531409746757265
# Optional - Each Pico-W will generate its own MAC address, so only add this if needed.
WiFiMACAddress=00:80:19:C0:FF:EE
---------------
```

```
-------------- log -------------
Platform: BlueSCSI
FW Version: 2024.05.21-rel May 21 2024 21:21:44
I2C Supported
Flash chip size: 2048 kB
 
=== SD Card Info ===
SD card detected, FAT32 volume size: 3773 MB
SD Name: SU04G, MID: 0x03, OID: 0x53 0x44
[175ms]  
[175ms] === Global Config ===
[176ms] Reading configuration from bluescsi.ini
[176ms] Active configuration (using system preset "Generic"):
[177ms] -- Debug is enabled
[177ms] DBG -- Parity is enabled
[177ms] DBG -- ReinsertCDOnInquiry is disabled
[225ms]  
[225ms] === Finding images in / ===
[226ms] == Opening /NE5.hda for ID: 5 LUN: 0
[226ms] ---- Configuring as network based on image name
[226ms] DBG ---- Read prefetch enabled: 8192 bytes
[227ms] ---- Image ready
[227ms] == Opening /HD1.hda for ID: 1 LUN: 0
[451ms] DBG ---- Read prefetch enabled: 8192 bytes
[451ms] ---- Image ready
[452ms] 
[452ms] === ROM Drive ===
[452ms] Platform supports ROM drive up to 1692 kB
[452ms] ---- ROM drive image not detected
[452ms]  
[452ms] === Configured SCSI Devices ===
[453ms] * ID: 1, BlockSize: 512, Type: Fixed, Quirks: None, Size: 1855500kB
[453ms] * ID: 5, Type: Network, Quirks: None
[1260ms]  
[1260ms] === Network Initialization ===
[1261ms] Unique board id: e6 61 30 10  0f 1b 7a 22
[2174ms] Wi-Fi MAC: 00:80:19:10:0F:1B
[2174ms] Connecting to Wi-Fi SSID "Fritz!Box Frank" with WPA/WPA2 PSK
---------------------------------
```

The log say it is connected but my Router say not ...!

I want to use it with my Atari TOS Computer but without connection by the bluescsi.ini must be work ...

**Comment by erichelgeson** [comment]
First I'd remove all the extra/unneeded config in the ini, just the WiFi ssid/password. You can try quoting both fields too, eg:
```ini
[SCSI]
WiFiSSID="Fritz!Box-Frank"
WiFiPassword="18266531409746757265"
```

Note the log you showed has 2 issues - one ssid nam is different than in you config, one is `Fritz!Box-Frank` and the other is `Fritz!Box Frank` - make sure you have it set correctly - two - doesn't say connected, it says connecting.

Let me know if that helps.

**Reply by franklukas** [reply]
I change the name to Fritz!Box-Frank and my ini take that.

I think I have a issuse with my Router (Fritzbox Fon Wlan 7360) and the BlueSCSI ...? I use at home the WiFi without problems with a MacMini 2012, Macbook Pro and a MobilePhone. 

The log from the BlueSCSI say the WiFi is connected but is no ...!

**Reply by franklukas** [reply]
```
------------- bluescsi.ini -------------
[SCSI]
WiFiSSID="Fritz!Box-Frank"
WiFiPassword="97565831136958511234"
WiFiMACAddress=00:80:19:C0:FF:EE
Debug=1
-------------------------------------
```
```
------------- log -------------------
Platform: BlueSCSI
FW Version: 2024.05.21-rel May 21 2024 21:21:44
Flash chip size: 2048 kB
 
=== SD Card Info ===
SD card detected, FAT32 volume size: 3773 MB
SD Name: SU04G, MID: 0x03, OID: 0x53 0x44
[190ms]  
[190ms] === Global Config ===
[194ms] Reading configuration from bluescsi.ini
[197ms] Active configuration:
[200ms] -- Debug is enabled
[202ms] DBG -- Parity is enabled
[205ms] DBG -- ReinsertCDOnInquiry is disabled
[244ms]  
[244ms] === Finding images in / ===
[246ms] == Opening /HD1.hda for ID: 1 LUN: 0
[481ms] DBG ---- Read prefetch enabled: 8192 bytes
[484ms] DBG ---- Valid Macintosh Device Image detected.
[488ms] ---- Image ready
[490ms] == Opening /NE5.txt for ID: 5 LUN: 0
[494ms] ---- Configuring as network based on image name
[499ms] DBG ---- Read prefetch enabled: 8192 bytes
[503ms] ---- Image ready
[506ms] 
[506ms] === ROM Drive ===
[508ms] Platform supports ROM drive up to 1692 kB
[513ms] ---- ROM drive image not detected
[516ms]  
[517ms] === Configured SCSI Devices ===
[521ms] * ID: 1, BlockSize: 512, Type: Fixed, Quirks: Apple, Size: 1855500kB
[527ms] * ID: 5, Type: Network, Quirks: Apple
[1339ms]  
[1339ms] === Network Initialization ===
[2253ms] Wi-Fi MAC: 00:80:19:C0:FF:EE
[2254ms] Connecting to Wi-Fi SSID "Fritz!Box-Frank" with WPA/WPA2 PSK
[2276ms]  
[2276ms] Initialization complete!
[2317ms] INFO: Pico Voltage: 3.276V.
-------------------------------------------
```

**Reply by erichelgeson** [reply]
The Pi Pico-w only supports 2.4Ghz networks -make sure you've selected that. I've not heard of any compatibility issues with the Pico-W and specific routers.

You have booted this in your Mac and setup MacTCP or opentransport and verified it does not work? You said the WiFi DA connects to the network and you can use it in your initial post, correct? If that's the case it's a config issue somewhere.

**Comment by franklukas** [comment]
Yes my Router works at 2,4Ghz ...

![Fritzbox](https://github.com/user-attachments/assets/bbf7fa07-2366-4371-841d-3d942ed0222b)


The log say it is connected but my Router say no! 

When I use the WiFi Accessory (https://jcs.org/bluescsi) I can connect to my Router and I can use a FTP Client so the MacTCP Setup is fine.

**Reply by franklukas** [reply]
... maybe the assumption is correct that it is a timeout problem?

**Reply by erichelgeson** [reply]
Thanks for confirming it's 2.4 - some routers don't turn on 2.4 by default.

> The log say it is connected but my Router say no!

Where does the log say that? The ones you've posted so far say "Connecting" not "Successfully connected" - I just want to make sure I'm not missing something.

You should see this type of log on success:
```
=== Network Initialization
WiFi MAC is 00:80:19:C0:FF:EE
Connecting to WiFi SSID "UniFi" with WPA/WPA2 PSK
Successfully connected to WiFi SSID UniFi
```

The WiFi DA from @jcs configures the same parameters so there is some parsing issue with the ini file or a typo or something. The Pico's WiFi library doesn't give us a ton of info on why it failed to connected. 

Note that if your password has `;` or `#` characters you'll need to quote them as they are comments for the ini file, though it looks like you have. I can try to setup an ssid with an `!` in it, though I dont see why that should matter (just guessing here as it seems like you have things setup correctly)

**Reply by franklukas** [reply]
```
-------- log ---------

Platform: BlueSCSI
FW Version: 2024.05.21-rel May 21 2024 21:21:44
Flash chip size: 2048 kB
 
=== SD Card Info ===
SD card detected, FAT32 volume size: 3773 MB
SD Name: SU04G, MID: 0x03, OID: 0x53 0x44
[183ms]  
[183ms] === Global Config ===
[187ms] Reading configuration from bluescsi.ini
[190ms] Active configuration:
[193ms] -- Debug is enabled
[195ms] DBG -- Parity is enabled
[198ms] DBG -- ReinsertCDOnInquiry is disabled
[241ms]  
[241ms] === Finding images in / ===
[243ms] == Opening /HD1.hda for ID: 1 LUN: 0
[467ms] DBG ---- Read prefetch enabled: 8192 bytes
[470ms] DBG ---- Valid Macintosh Device Image detected.
[473ms] ---- Image ready
[476ms] == Opening /NE5.txt for ID: 5 LUN: 0
[480ms] ---- Configuring as network based on image name
[484ms] DBG ---- Read prefetch enabled: 8192 bytes
[489ms] ---- Image ready
[492ms] 
[492ms] === ROM Drive ===
[494ms] Platform supports ROM drive up to 1692 kB
[498ms] ---- ROM drive image not detected
[502ms]  
[503ms] === Configured SCSI Devices ===
[506ms] * ID: 1, BlockSize: 512, Type: Fixed, Quirks: Apple, Size: 1855500kB
[513ms] * ID: 5, Type: Network, Quirks: Apple
[1325ms]  
[1325ms] === Network Initialization ===
[2239ms] Wi-Fi MAC: 00:80:19:C0:FF:EE
[2240ms] Connecting to Wi-Fi SSID "Fritz!Box-Frank" with WPA/WPA2 PSK
[2262ms]  
[2262ms] Initialization complete!
[2296ms] INFO: Pico Voltage: 3.265V.
----------------------
```

I deleted the log and restarted the computer. When I start fetch ftp the BlueSCSI activity LED flashes continuously and then stays on for a short time without flashing and then it starts flashing again from the beginning and so on ...

```
------------ ini ----
[SCSI]
WiFiSSID="Fritz!Box-Frank"
WiFiPassword="31417231136958518620"
WiFiMACAddress=00:80:19:C0:FF:EE
Debug=1
```

**Reply by franklukas** [reply]
Can you send me a debug firmware version to find out the fault ...?


On the Macintosh, if the connection cannot be established from the ini, the activity LED flashes for a while, then it lights up continuously for a short time and then it starts again from the beginning with flashes and so on ...

On the Atari Falcon, if the connection cannot be established, the activity LED lights up continuously without going out and MiNT hangs in the boot process and nothing works except a hardware reset on the computer. With a real Dayna SCSILINK all is fine.

![mint_boot](https://github.com/user-attachments/assets/2ad70562-208e-4920-8888-e359faa386ed)

**Reply by franklukas** [reply]
I use the 25 D-Sub BlueSCSI version with my Mac IIsi and the Wi-Fi works but not so good and not without problems ... . Connect over the ini at the first boot do not work. When I use the Wi-Fi Accessory I can connect to my Router and when I make a reboot the connecting over the ini works ...!

With the Desktop BlueSCSI Version on my Atari Falcon the Wi-Fi do not work at all. I use MiNT and it stuck at the boot process, the activity LED lights up permanently and the entire system hangs ...!

**Reply by erichelgeson** [reply]
>  when I make a reboot the connecting over the ini works ...!

If the bluescsi wasn't power cycled then it is still connected to the wifi from the DA.

I suppose I could make a firmware that logs out the hex values of the WiFi config values just to validate it's the same bytes, but the connection function is the same regardless if it's via the DA or the ini. I'll try to get to that later today.

I just want to mention for this issue or even the other Discussion you have that Troy is helping you with - please do go through the Troubleshooting guide as it can fix many odd issues https://github.com/BlueSCSI/BlueSCSI-v2/wiki/Troubleshooting

**Reply by erichelgeson** [reply]
[firmware.uf2.zip](https://github.com/user-attachments/files/16285006/firmware.uf2.zip)

Please upload the full debug log of this - you should have your ini config set, boot the mac, validate you are not connected to the wifi, use the wifi da to connect, validate you are connected. shutdown. grab the log file (will be quite big) and provide it. Only change this firmware has is logging out the full ssid and password as hex. 

And to state the obvious, your ssid and password will be seen in this log - if you'd like to send it to me privately my email is easy to find online.

**Reply by franklukas** [reply]
I'm not sure if I understood everything correctly...


25 D-Sub BlueSCSI v2 firmware update over USB. Mac IIsi boot up and connect over the DA to my Router and Shutdown ...

[log.txt](https://github.com/user-attachments/files/16285563/log.txt)

bluescsi.ini ...
[SCSI]
System=Generic
EnableSCSI2=0 ; Off
EnableParity=0 ; Off
Debug=1
WiFiSSID="Fritz!Box-Frank"
WiFiPassword="password"
WiFiMACAddress=00:80:19:C0:FF:EE

**Reply by erichelgeson** [reply]
Ok perfect! I think I see the issue, will get you the full answer this afternoon (my time, -6GMT)

**Reply by erichelgeson** [reply, marked as answer]
The issue is that the SSID is incorrect in the `ini` file.

 SSID's are case sensitive - You can see here that the ini has `FRITZ` in all caps, and the WiFi DA has `Fritz`. 

Once you address that the ini should work for you!

```
// Log from ini attempt to connect:
Connecting to Wi-Fi SSID "FRITZ!Box-Frank" with WPA/WPA2 PSK
SSID:
 46 F - ok
 52 Issue: a capital R not a lower case r like below
 49 Issue: same but I
 54 Issue: same but T
 5a Issue: same but Z
 21 !
 42 B
 6f o
 78 x
 2d -
 46 F
 72 r
 61 a
 6e n
 6b k
```
```
// Log from WiFi DA connect:
Connecting to Wi-Fi SSID "Fritz!Box-Frank" with WPA/WPA2 PSK
SSID:
 46 F
 72 r
 69 i
 74 t
 7a z
 21 !
 42 B
 6f o
 78 x
 2d -
 46 F
 72 r
 61 a
 6e n
 6b k
```
I'll note again you don't need to add the `WiFiMACAddress` (if you have 2 bluescsi's on the same network you'll have issues as you're using the same MAC Address)

**Reply by franklukas** [reply]
Ok nice find but what should I do about it ...?

I can rename my SSID to a couple of numbers or can I use hex numbers in the ini ?

**Reply by erichelgeson** [reply]
Are you 100% sure that your ini file has `WiFiSSID="Fritz!Box-Frank"` in it? BlueSCSI is not chaining the case of just partial words, it is only reading in the text in the file. I would do a full overwrite format of the SD card to ensure nothing is left over and recreate the ini file. Also make sure you are showing file extensions and there's not a `bluescsi.ini.txt` or something that you're editing instead of `bluescsi.ini`

edit: I just tried exactly `WiFiSSID="Fritz!Box-Frank"` and the hex dumped was correct from the ini.

**Reply by franklukas** [reply]
Thank you for your Help.


The problem is in front of the computer...!

The SSID name in my router config was actually capitalized as "FRITZ!Box-Frank"..., so all was fine.

**Reply by erichelgeson** [reply]
No problem! Glad it's simple :) Enjoy!

**Reply by franklukas** [reply]
The problem with the Atari remains...

With a real Dayna SCSILINK everything works fine. With the BlueSCSI v2 Desktop Version and the right SSID name the System hangs not longer but at booting time it come up an error and I must reboot the machine. The SSID in the ini is right so the activity LED no longer lights up permanently. I'll ask in the Atari forum first...


![MiNT_Error](https://github.com/user-attachments/assets/e54f9ab8-23d2-45df-8c89-07095c7e6a5a)

**Reply by franklukas** [reply]
Maybe also need help and advice from the Bluescsi developers...?

https://github.com/freemint/freemint/issues/342


The original Dayna SCSILINK works fine with MiNT so the BlueSCSI Wi-Fi maybe it needs slightly different parameters in the Atari driver ...?

**Reply by erichelgeson** [reply]
Our emulation is based off observations/captures of the official DaynaPORT driver on a Macintosh - I know there was some 3rd party drivers for other systems created but they did not behave the same way as the official driver. If we can understand those differences we can probably evaluate the feasibility of accommodating that here or adjusting the driver there.

**Reply by franklukas** [reply]
it would be very nice if you could post in the Atari thread and help to solve the problem ...

https://github.com/freemint/freemint/issues/342

**Reply by erichelgeson** [reply]
Unfortunately I don't know anything about freemint or about the driver, so I cant be much help now. I'll watch the thread and chime in if there's anything I can add value to on the BlueSCSI side of things.
